### PR TITLE
feat(docs): document `events` package

### PR DESCRIPTION
**Pull request by pierwill**

### CODE REVIEW

File: stdlib/contrib/tomhollingworth/events/duration.flux [hunk 1]
```diff
@@ -1,16 +1,77 @@
...
+//   Default is `1ns`
+// - columnName: Name of the result column.
+//   Default is `"duration"`.
+// - timeColumn: Name of the time column.
+//   Default is `"_time"`.
+// - stopColumn: Name of the stop column.
+//   Default is `"_stop"`.
+// - stop: The latest time to use when calculating results.
+//   If provided, `stop` overrides the time value in the `stopColumn`.
+//
```
Comment (sanderson) [inline stdlib/contrib/tomhollingworth/events/duration.flux:34]:
```suggestion
// - stop: The latest time to use when calculating results.
//
//   If provided, `stop` overrides the time value in the `stopColumn`.
//
// - tables: Input data. Default is piped-forward data (`<-`).
//
```

File: stdlib/contrib/tomhollingworth/events/duration.flux [hunk 2]
```diff
@@ -1,16 +1,77 @@
...
+// ### Calculate the duration of states
+//
+// ```
+// import "contrib/tomhollingworth/events"
+//
+// data
+//   |> events.duration(
+//     unit: 1m,
+//     stop: 2020-01-02T00:00:00Z
+//   )
```
Comment (sanderson) [inline stdlib/contrib/tomhollingworth/events/duration.flux:45]:
```suggestion
// import "array"
// import "contrib/tomhollingworth/events"
// 
// # data = array.from(
// #     rows: [
// #         {_time: 2020-01-01T00:00:00Z, state: "ok"},
// #         {_time: 2020-01-01T00:12:34Z, state: "warn"},
// #         {_time: 2020-01-01T00:25:01Z, state: "ok"},
// #         {_time: 2020-01-01T16:07:55Z, state: "crit"},
// #         {_time: 2020-01-01T16:54:21Z, state: "warn"},
// #         {_time: 2020-01-01T18:20:45Z, state: "ok"},
// #     ],
// # )
// #
// < data
//     |> events.duration(
//         unit: 1m,
//         stop: 2020-01-02T00:00:00Z,
// >     )
```

File: stdlib/contrib/tomhollingworth/events/duration.flux [hunk 3]
```diff
@@ -1,16 +1,77 @@
...
+//   unit: 1m
+// )
+//
+// data |> stateDuration(
+//   unit: 1m,
+//   fn: (r) => true
+// )
+// ```
+//
+// [OUTPUT]
```
Comment (sanderson) [inline stdlib/contrib/tomhollingworth/events/duration.flux:74]:
~~~suggestion
// ### Compared to similar functions
//
// The example below includes output values of
// `events.duration()`, `elapsed()`, and `stateDuration()`
// related to the `_time` and `state` values of input data.
//
// ```
// import "array"
// import "contrib/tomhollingworth/events"
//
// # data = array.from(
// #     rows: [
// #         {_time: 2020-01-01T00:00:00Z, state: "ok"},
// #         {_time: 2020-01-01T00:12:34Z, state: "warn"},
// #         {_time: 2020-01-01T00:25:01Z, state: "ok"},
// #         {_time: 2020-01-01T16:07:55Z, state: "crit"},
// #         {_time: 2020-01-01T16:54:21Z, state: "warn"},
// #         {_time: 2020-01-01T18:20:45Z, state: "ok"},
// #     ],
// # )
// #
// union(
//     tables: [
//         data |> events.duration(unit: 1m, stop: 2020-01-02T00:00:00Z) |> map(fn: (r) => ({_time: r._time, state: r.state, function: "events.Duration()", value: r.duration})),
//         data |> elapsed(unit: 1m) |> map(fn: (r) => ({_time: r._time, state: r.state, function: "elapsed()", value: r.elapsed})),
//         data |> stateDuration(unit: 1m, fn: (r) => true) |> map(fn: (r) => ({_time: r._time, state: r.state, function: "stateDuration()", value: r.stateDuration})),
//     ],
// )
// >     |> pivot(rowKey: ["_time", "state"], columnKey: ["function"], valueColumn: "value")
// ```
//
// tags: transformations,events
//
~~~

**Review by sanderson — CHANGES_REQUESTED** [review]

File: stdlib/contrib/tomhollingworth/events/duration.flux [hunk 4]
```diff
@@ -1,16 +1,98 @@
...
+// the function compares the timestamp of the final record
+// to the timestamp in the stopColumn or the specified stop time.
+//
+// > ## Similar functions
+// > events.duration() is similar to elapsed() and stateDuration(), but differs in important ways:
+// >
+// >elapsed() drops the first record. events.duration() does not.
+// >stateDuration() calculates the total time spent in a state (determined by a predicate function). events.duration() returns the duration between all records and their subsequent records.
+// >
+// >For examples, see below.
```
Comment (sanderson) [inline stdlib/contrib/tomhollingworth/events/duration.flux:21]: - This stuff will just get dropped into the JSON output as raw markdown, so we need to format code as code.
- We won't be using blockquote styling for notes in the generated docs.

```suggestion
// ### Similar functions
// `events.duration()` is similar to `elapsed()` and `stateDuration()`, but differs in important ways:
// 
// - `elapsed()` drops the first record. `events.duration()` does not.
// - `stateDuration()` calculates the total time spent in a state (determined by a predicate function).
//   `events.duration()` returns the duration between all records and their subsequent records.
// 
// See the example [below](#compared-to-similar-functions).
```

File: stdlib/contrib/tomhollingworth/events/duration.flux [hunk 5]
```diff
@@ -1,16 +1,98 @@
...
-// `columnName` - Optional string. Default 'duration'. The name of the result column
-// `stopColumn` - Optional string. Default '_stop'. The name of the column to compare the last record on
-// `stop` - Optional Time. Use a fixed time to compare the last record against instead of stop column.
+// duration calculates the duration of events.
+//
+// The function determines the time between a record and the subsequent record
+// and associates the duration with the first record (start of the event).
+// To calculate the duration of the last event,
+// the function compares the timestamp of the final record
+// to the timestamp in the stopColumn or the specified stop time.
```
Comment (sanderson) [inline stdlib/contrib/tomhollingworth/events/duration.flux:13]:
```suggestion
// to the timestamp in the `stopColumn` or the specified stop time.
```

**Review by sanderson — APPROVED** [review]
Two suggestions and it's gtg.

**Review by jsternberg — APPROVED** [review]
Just needs a rebase and to regenerate files.

**Pull request merged**